FraihmworkHealthAndIntegrityApi.yaml

```diff
@@ -795,6 +795,203 @@ paths:
+  /monitor/v1/component/schedule/unavailability/add:
+    post:
+      tags:
+      - Components
+      summary: Add unavailability schedules for one or more components.
+      description: |
+        Add unavailability schedules for one or more components.
+        - You can specify schedules for multiple components in one request.
+        - Multiple schedules can be added in a single request by providing an array of `UnavailabilitySchedule` objects.
+        - Schedules can specify specific days or use "All" to represent every day of the week.
+        - Users can also post separate days, such as "Monday" in addition to "All," to provide more control over the unavailability schedule for specific days.
+      operationId: addUnavailabilityScheduleUsingPOST
+      requestBody:
+        description: The unavailability schedule data for one or more components
+        required: true
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/UnavailabilitySchedule'
+      responses:
+        201:
+          description: Successfully added the unavailability schedule(s)
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        400:
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        401:
+          description: Not authenticated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        403:
+          description: Forbidden - client is not authorized to add the schedule
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        406:
+          description: Not acceptable - only responds in application/json format
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        500:
+          description: Server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+      security:
+      - OAuth2: [fraihmwork.component.write, fraihmwork.component.admin]
+
+  /monitor/v1/component/schedule/unavailability:
+    get:
+      tags:
+      - Components
+      summary: Get the unavailability schedule for one or more components.
+      description: |
+        Retrieve the unavailability schedule for one or more components.
+        - Provide one or more `componentId` values as query parameters to filter the results.
+        - If no `componentId` is provided, schedules for all components will be retrieved.
+        - The response will return a separate schedule for each component.
+        - Within each component's schedule:
```

Comment:
I'm uncertain how the response shape (an array of UnavailabilitySchedules would be able to group similar days here.  Can we add some [examples](https://swagger.io/specification/) in the spec to make it clearer?

My rough guess here on what the response looks like:
```
[
  {
    componentId: "123",
    schedules: [
       days: [ Monday, Tuesday ],
       schedules: [ 9-10am, 5pm-12am]
    ]
  },
  {
    componentId: "456",
    schedules: [
       days: [ Monday, Tuesday ],
       schedules: [ 9-10am, 5pm-12am]
    ]
  }
]
```

Reply:
Yes, your guess aligns with what i was thinking as well. I have updated the GET API spec to include examples for both single and multiple components

Here is the structure I've added to the spec:


For a single component, we see that Monday has a specific time period that is not shared with other days. Then, Tuesday, Thursday, Friday, and Saturday share the same start time and end time so they are grouped together.. Wednesday and Sunday do not have any unavailability periods specified, so they are also grouped together with empty timePeriods.


`[
  {
    "componentId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
    "schedules": [
      {
        "days": [
          "Monday"
        ],
        "timePeriods": [
          {
            "startTime": "14:00",
            "endTime": "23:59"
          }
        ]
      },
      {
        "days": [
          "Tuesday",
          "Thursday",
          "Friday",
          "Saturday"
        ],
        "timePeriods": [
          {
            "startTime": "17:00",
            "endTime": "23:59"
          }
        ]
      },
      {
        "days": [
          "Wednesday",
          "Sunday"
        ],
        "timePeriods": []
      }
    ]
  }
]
`


Let me know if this example makes sense. (also added this into api spec )
![image](https://github.com/user-attachments/assets/88aceb02-ca52-4527-b8b9-a647ff885c1a)


Also updated description that the order in which days and schedules are returned is not guaranteed and should not be relied upon.